src/solvers/dgmulti/flux_differencing_compressible_euler.jl

```diff
@@ -11,180 +11,169 @@
 #       loop optimally when passing our custom functions `cons2entropy` and
 #       `entropy2cons`. Thus, we need to insert the physics directly here to
 #       get a significant runtime performance improvement.
-function entropy_projection!(cache, u::StructArray{<:SVector{4, <:Union{Float32, Float64}}},
-                             mesh::VertexMappedMesh, equations::CompressibleEulerEquations2D, dg::DGMultiFluxDiff{<:Polynomial})
-  rd = dg.basis
-  @unpack Vq = rd
-  @unpack VhP, u_values, entropy_var_values = cache
-  @unpack projected_entropy_var_values, entropy_projected_u_values = cache
-
-  apply_to_each_field(mul_by!(Vq), u_values, u)
-
+function cons2entropy!(entropy_var_values::StructArray,
+                       u_values          ::StructArray,
+                       equations::CompressibleEulerEquations2D)
   # The following is semantically equivalent to
-  # @threaded for i in Base.OneTo(length(u_values))
+  # @threaded for i in eachindex(u_values)
   #   entropy_var_values[i] = cons2entropy(u_values[i], equations)
   # end
   # but much more efficient due to explicit optimization via `@turbo` from
   # LoopVectorization.jl.
   @unpack gamma, inv_gamma_minus_one = equations
-  let
-    rho_values, rho_v1_values, rho_v2_values, rho_e_values = StructArrays.components(u_values)
-    w1_values, w2_values, w3_values, w4_values = StructArrays.components(entropy_var_values)
-
-    @turbo thread=true for i in indices(
-        (rho_values, rho_v1_values, rho_v2_values, rho_e_values,
-         w1_values, w2_values, w3_values, w4_values))
-      rho    = rho_values[i]
-      rho_v1 = rho_v1_values[i]
-      rho_v2 = rho_v2_values[i]
-      rho_e  = rho_e_values[i]
-
-      # The following is basically the same code as in `cons2entropy`
-      v1 = rho_v1 / rho
-      v2 = rho_v2 / rho
-      v_square = v1^2 + v2^2
-      p = (gamma - 1) * (rho_e - 0.5 * rho * v_square)
-      s = log(p) - gamma * log(rho)
-      rho_p = rho / p
-
-      w1_values[i] = (gamma - s) * inv_gamma_minus_one - 0.5 * rho_p * v_square
-      w2_values[i] = rho_p * v1
-      w3_values[i] = rho_p * v2
-      w4_values[i] = -rho_p
-    end
-  end
 
-  # "VhP" fuses the projection "P" with interpolation to volume and face quadrature "Vh"
-  apply_to_each_field(mul_by!(VhP), projected_entropy_var_values, entropy_var_values)
+  rho_values, rho_v1_values, rho_v2_values, rho_e_values = StructArrays.components(u_values)
+  w1_values, w2_values, w3_values, w4_values = StructArrays.components(entropy_var_values)
```

Comment:
These seem to be the only lines in the LV-specialized code which explicitly depend on `StructArrays`. If that's true, we could also extend these LV-optimizations of compressible Euler to more general array types by dispatching on a `get_components` command. This would then enable LV acceleration of `DGMultiFluxDiff{<:SBP}` (e.g., DGSEM) too, which uses `Matrix{<:SVector}` solution arrays.

What do you think? This would be a separate PR, of course.

Reply:
I'm not sure whether it wouldn't be better to avoid the additional pointer indirection of creating `view`s to for `get_components` instead of working with a reinterpreted array directly. Do you something like `cons2entropy!` for `DGMultiFluxDiff{<:SBP}`?

Reply:
Ah, that's right! We don't - never mind this comment then.